src/error.rs

```diff
@@ -320,99 +404,6 @@ pub enum ErrorKind {
     WriteError(WriteFailure),
```

Comment:
I left `ErrorKind::SessionsNotSupported` without an "error" prefix since it doesn't have any error data associated with it. Let me know if that seems weird or if it makes sense.

Reply:
I would say feels a little more consistent to me to add on the "Error" at the end, because it seems like the enum instance itself is an error, whether or not the case has associated values (or whatever the rust term is for those)

Reply:
Resolving per discussion above.